streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -887,27 +887,27 @@ void process(final ProcessorSupplier<? super K, ? super V> processorSupplier,
      * "-repartition" is a fixed suffix.
      * You can retrieve all generated internal topic names via {@link KafkaStreams#toString()}.
      * <p>
-     * All data of this stream will be redistributed through the repartitioning topic by writing all records to it,
+     * All data of this (input) stream will be redistributed through the repartitioning topic by writing all records to it,
      * and rereading all records from it, such that the resulting {@link KGroupedStream} is partitioned on the new key.
      * <p>
      * This is equivalent to calling {@link #selectKey(KeyValueMapper)} followed by {@link #groupByKey(Serde, Serde)}.
      *
-     * @param selector a {@link KeyValueMapper} that computes a new key for grouping
-     * @param keySerde key serdes for materializing this stream,
-     *                 if not specified the default serdes defined in the configs will be used
-     * @param valSerde value serdes for materializing this stream,
-     *                 if not specified the default serdes defined in the configs will be used
-     * @param <KR>     the key type of the result {@link KGroupedStream}
+     * @param selector             a {@link KeyValueMapper} that computes a new key for grouping
+     * @param inputKStreamKeySerde key serdes for materializing this (input) stream,
+     *                             if not specified the default serdes defined in the configs will be used
+     * @param inputKStreamValSerde value serdes for materializing this (input) stream,
+     *                             if not specified the default serdes defined in the configs will be used
+     * @param <KR>                 the key type of the result {@link KGroupedStream}
      * @return a {@link KGroupedStream} that contains the grouped records of the original {@code KStream}
      * @see #groupByKey()
      */
     <KR> KGroupedStream<KR, V> groupBy(final KeyValueMapper<? super K, ? super V, KR> selector,
-                                       final Serde<KR> keySerde,
-                                       final Serde<V> valSerde);
+                                       final Serde<KR> inputKStreamKeySerde,
```

Comment:
These names aren't correct as the Serdes here are for the result of the `groupBy`, so they are not the output. Generally i think i'd prefer the params to stay as `keySerde` and `valSerde` and just explain what they are in the javadoc